codex/node.nim

```diff
@@ -56,17 +59,47 @@ type
     validator: ?ValidatorInteractions
 
   CodexNodeRef* = ref object
-    switch*: Switch
-    networkId*: PeerId
-    blockStore*: BlockStore
-    engine*: BlockExcEngine
-    erasure*: Erasure
-    discovery*: Discovery
+    switch: Switch
+    networkId: PeerId
+    blockStore: BlockStore
+    engine: BlockExcEngine
+    erasure: Erasure
+    discovery: Discovery
     contracts*: Contracts
     clock*: Clock
 
   OnManifest* = proc(cid: Cid, manifest: Manifest): void {.gcsafe, closure.}
 
+func switch*(self: CodexNodeRef): Switch =
+  return self.switch
```

Comment:
Why add accessors? Do you think we'll need accessors in the future or is it just to prevent writes?

Reply:
Prevent writes